warp10/src/main/java/io/warp10/script/functions/HHCODEFUNC.java

```diff
@@ -0,0 +1,135 @@
+
+    Object o = stack.pop();
+
+    if (!(o instanceof Long)) {
+      throw new WarpScriptException(getName() + " expects resolution (even number between 2 and 30).");
+    }
+
+    int res = ((Number) o).intValue();
+
+    if (0 != res && (res < 2 || res > 30 || (0 != (res & 1)))) {
+      throw new WarpScriptException(getName() + " expects a maximum resolution which is an even number between 2 and 30 or 0.");
+    }
+
+
+    Object hhcode = stack.pop();
+
+    long hh;
+    boolean isLongFormat = true;
+
+    if (hhcode instanceof Long) {
+      hh = (long) hhcode;
+    } else if (hhcode instanceof String) {
+      String hhstr = hhcode.toString();
+      if (hhstr.length() > 16) {
+        throw new WarpScriptException(getName() + " expects an hexadecimal HHCode string of length <= 16");
+      } else if (hhstr.length() < 16) {
+        hhcode = new StringBuilder(hhstr).append("0000000000000000");
+        ((StringBuilder) hhcode).setLength(16);
+      }
+      hh = Long.parseUnsignedLong(hhcode.toString(), 16);
+      isLongFormat = false;
+    } else if (hhcode instanceof byte[]) {
+      hh = Longs.fromByteArray((byte[]) hhcode);
+    } else {
+      throw new WarpScriptException(getName() + " expects a long, a string or a byte array.");
+    }
+
+    switch (this.action) {
+      case NORTH:
+        stack.push(this.manageFormat(HHCodeHelper.northHHCode(hh, res), res, isLongFormat));
+        break;
+      case SOUTH:
+        stack.push(this.manageFormat(HHCodeHelper.southHHCode(hh, res), res, isLongFormat));
+        break;
+      case EAST:
+        stack.push(this.manageFormat(HHCodeHelper.eastHHCode(hh, res), res, isLongFormat));
+        break;
+      case WEST:
+        stack.push(this.manageFormat(HHCodeHelper.westHHCode(hh, res), res, isLongFormat));
+        break;
+      case NORTH_EAST:
+        stack.push(this.manageFormat(HHCodeHelper.northEastHHCode(hh, res), res, isLongFormat));
+        break;
+      case NORTH_WEST:
+        stack.push(this.manageFormat(HHCodeHelper.northWestHHCode(hh, res), res, isLongFormat));
+        break;
+      case SOUTH_EAST:
+        stack.push(this.manageFormat(HHCodeHelper.southEastHHCode(hh, res), res, isLongFormat));
+        break;
+      case SOUTH_WEST:
+        stack.push(this.manageFormat(HHCodeHelper.southWestHHCode(hh, res), res, isLongFormat));
+        break;
+      case BBOX:
+        double[] bbox = HHCodeHelper.getHHCodeBBox(hh, res);
+        stack.push(bbox[0]);
+        stack.push(bbox[1]);
+        stack.push(bbox[2]);
+        stack.push(bbox[3]);
+        break;
+      case CENTER:
+        double[] latlon = HHCodeHelper.getCenterLatLon(hh, res);
+        stack.push(latlon[0]);
+        stack.push(latlon[1]);
+        break;
+      default:
+        throw new WarpScriptException("Unknown HHCODE action");
+    }
+
+    return stack;
+  }
+
+  private Object manageFormat(long hh, int res, boolean isLongFormat) {
```

Comment:
Actually since you accept either LONG, STRING or BYTES as input it would be more logical to output a byte array if the input was a byte array. So instead of passing isLongFormat, why not simply pass the input and check its type in manageFormat?

Reply:
done